ledger/error/generator/app/src/main/scala/com/daml/error/generator/app/ErrorCodeInventoryDocsGenApp.scala

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.error.generator.app
+
+import java.nio.file.{Files, Paths, StandardOpenOption}
+
+import com.daml.error.{ErrorClass, Grouping}
+import com.daml.error.generator.{ErrorCodeDocumentationGenerator, ErrorDocItem, GroupDocItem}
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+/** Generates error codes inventory as a reStructuredText
+  */
+object ErrorCodeInventoryDocsGenApp {
+
+  def main(args: Array[String]): Unit = {
+    val text = {
+      val (errorDocItems, groupDocItems): (Seq[ErrorDocItem], Seq[GroupDocItem]) =
+        new ErrorCodeDocumentationGenerator().getDocItems
+
+      val groupSegmentsToExplanationMap: Map[List[Grouping], Option[String]] =
+        groupDocItems.map { groupDocItem: GroupDocItem =>
+          groupDocItem.errorClass.groupings -> groupDocItem.explanation.map(_.explanation)
+        }.toMap
+
+      val errorCodes: Seq[ErrorCodeValue] = errorDocItems.map { (errorDocItem: ErrorDocItem) =>
+        ErrorCodeValue(
+          category = errorDocItem.category,
+          errorGroupPath = errorDocItem.hierarchicalGrouping,
+          conveyance = errorDocItem.conveyance.getOrElse("").replace('\n', ' '),
+          code = errorDocItem.code,
+          deprecationO = errorDocItem.deprecation.map(_.deprecation.replace('\n', ' ')),
+          explanation = errorDocItem.explanation.fold("")(_.explanation).replace('\n', ' '),
+          resolution = errorDocItem.resolution.fold("")(_.resolution).replace('\n', ' '),
+        )
+      }
+
+      val root = ErrorGroupTree.empty()
+
+      // Build trie like structure of error groups and error codes.
+      errorCodes.foreach(errorCode =>
+        root.insertErrorCode(errorCode, groupSegmentsToExplanationMap)
+      )
```

Comment:
Wouldn't a simple lexicographical ordering work just as good, while also being sorted alphabetically?

Reply:
Thought about that, but couldn't see how would I then calculate the hierarchical numberings (e.g. 1.2.2.1) without resorting to a tree structure anyway

Reply:
Ok. I still think this can be made simpler. But let's not block this PR on it 👍